source/agora/node/Registry.d

```diff
@@ -822,18 +824,28 @@ private struct ZoneData
             "FOREIGN KEY(pubkey) REFERENCES registry_%s_utxo(pubkey) ON DELETE CASCADE, " ~
             "PRIMARY KEY(pubkey, address))", zone_name, zone_name);
 
+        // Initialize common fields
         this.db.execute(query_sig_create);
         this.db.execute(query_addr_create);
 
-        if (this.type == ZoneType.primary)
-        {
-            // Serial's value wraps around so the cast is safe
-            const serial = cast(uint) Clock.currTime(UTC()).toUnixTime();
-            this.soa = this.config.fromConfig(this.root, serial);
-        }
-        else
-            // SOA is not default-constructible, so the compiler complains
-            this.soa = SOA.init;
+        this.soa = this.config.fromConfig(this.root);
```

Comment:
This is what is causing the test failure. If the zone is not primary, we shouldn't read the SOA from the config file.

Reply:
That would be the next cause, but currently main cause is `ZoneType` only includes `primary` and that's the default one.

Reply:
So, I needed to disable `post...` endpoint if Zone is not primary and test needs to be updated to initialize a primary Validator zone.

Reply:
I'm not sure why ? We can fwd `post`, no ?

Reply:
Secondary is not fully implemented in this PR, this only makes it easier to detect primary and non-primary zones through type field. Thus there is no fwd configuration available. Main reason behind is me allowing `onAcceptedBlock` to only `primary` zone, if I revert that, then there is no need to disable `post...`s

Reply:
Yeah that makes sense.